WordPress/gutenberg
# [RNMobile] Limit Ordered List Start Value to 32 bit Integer

## What?

Limits the start value for ordered lists to fit in a 32 bit integer: `2147483647`. 
This also limits the start value passed to the react native `<RichText>` component. 

## Why?

This fixes #39727
The mobile editor on Android is crashing when the start value is larger than an 32 bit integer. iOS can handle the larger values but the UI becomes unstable when the start value is over a 64 bit integer.

## How?

I've added two `Math.min(start, MAX_INT)` calls where the `ol` start value is handled in the `ol` list settings and in the props list that is passed to the native `<RichText>` component

## Testing Instructions




1. Open the mobile editor on Android
2. Insert an list block configured as an ordered list
3. Open the list block settings and enter a start value of `2147483647`
4. Verify that the value is allowed
5. Enable 'Reverse list numbering' in the ordered list settings
6. Close the options and verify that the ordered list starts at `2147483647`
7. Open the block settings and enter a value *larger* than `2147483647`
8. Notice that the value resets back to `2147483647`
9. Close the block setting and verify that the editor does not crash 

## Screenshots or screencast
